[
    {
        "content": "<p>Hi everyone,<br>\nthe publishing process of my IG fails with the following message:</p>\n<p>onGenerate:<br>\nPublishing Content Failed: null                                                  (00:15.0482)<br>\n                                                                                 (00:15.0486)<br>\nUse -? to get command line help                                                  (00:15.0489)<br>\n                                                                                 (00:15.0493)<br>\nStack Dump (for debugging):                                                      (00:15.0496)<br>\njava.lang.NullPointerException<br>\n        at org.hl7.fhir.utilities.xhtml.XhtmlNode.addTag(XhtmlNode.java:144)<br>\n        at org.hl7.fhir.utilities.xhtml.XhtmlNode.para(XhtmlNode.java:527)<br>\n        at org.hl7.fhir.r5.renderers.BundleRenderer.render(BundleRenderer.java:299)<br>\n        at org.hl7.fhir.igtools.publisher.Publisher.getXhtml(Publisher.java:7517)<br>\n        at org.hl7.fhir.igtools.publisher.Publisher.saveDirectResourceOutputs(Publisher.java:7008)<br>\n        at org.hl7.fhir.igtools.publisher.Publisher.generateHtmlOutputs(Publisher.java:6470)<br>\n        at org.hl7.fhir.igtools.publisher.Publisher.generate(Publisher.java:4775)<br>\n        at org.hl7.fhir.igtools.publisher.Publisher.createIg(Publisher.java:837)<br>\n        at org.hl7.fhir.igtools.publisher.Publisher.execute(Publisher.java:683)<br>\n        at org.hl7.fhir.igtools.publisher.Publisher.main(Publisher.java:8022)</p>\n<p>The cause appears to be the result attribute in the DiagnosticReport example  <a href=\"user_uploads/10155/7KAIVuiGkA7yYEYqTjF5BYug/IEHR_REPORT_SINGLE_IMAGE__RES_RadObservation.json\">IEHR_REPORT_SINGLE_IMAGE__RES_RadObservation.json</a><br>\nBecause when the following lines are removed from the example, the publisher works without issues.</p>\n<div class=\"codehilite\"><pre><span></span><code>  &quot;result&quot;: [ \n  {\n    &quot;reference&quot;: &quot;Observation/11&quot;\n  } \n  ],\n</code></pre></div>\n\n\n<p>Does anyone know how to fix this?</p>\n<p>Thanks for your attention,<br>\nNiklas<br>\nEdit: changed the Observation ID to fit the reference</p>",
        "id": 209093223,
        "sender_full_name": "Niklas Haldorn",
        "timestamp": 1599226043
    },
    {
        "content": "<p>Does Observation/11 exist and resolve?  (If it doesn't, it shouldn't blow up the publisher, but that might be the issue.)</p>",
        "id": 209093552,
        "sender_full_name": "Lloyd McKenzie",
        "timestamp": 1599226210
    },
    {
        "content": "<p>I mixed up the ids in the example, but I still get the error after changing the id to fit the reference</p>",
        "id": 209094272,
        "sender_full_name": "Niklas Haldorn",
        "timestamp": 1599226681
    },
    {
        "content": "<p><span class=\"user-mention\" data-user-id=\"191316\">@Grahame Grieve</span></p>",
        "id": 209097100,
        "sender_full_name": "Lloyd McKenzie",
        "timestamp": 1599228227
    },
    {
        "content": "<p>that's an error in my error handling code. I fixed that but I can't tell what the underlying problem is</p>",
        "id": 209136522,
        "sender_full_name": "Grahame Grieve",
        "timestamp": 1599250544
    },
    {
        "content": "<p>I get a NullPointerException with FHIR IG Publisher Version 1.1.53 (Git# bc0c1360906c). </p>\n<p>...<br>\nFinal .zip built                                                                 (06:28.0405)<br>\nFinished. Times: loading: 00:01.0772, generate: 05:42.0577, narrative generation: 00:23.0253, realm-rules: 00:00.0000, previous-ver<br>\nsion: 00:00.0000, jekyll: 03:17.0874, validation: 00:07.0983 (#34), template: 00:03.0448 (#3) (06:28.0840)<br>\nPublishing Content Failed: null                                                  (06:29.0058)<br>\n                                                                                 (06:29.0059)<br>\njava.lang.NullPointerException<br>\n        at org.hl7.fhir.igtools.renderers.ValidationPresenter.genHeader(ValidationPresenter.java:563)<br>\n        at org.hl7.fhir.igtools.renderers.ValidationPresenter.genQAHtml(ValidationPresenter.java:245)<br>\n        at org.hl7.fhir.igtools.renderers.ValidationPresenter.generate(ValidationPresenter.java:183)<br>\n        at org.hl7.fhir.igtools.publisher.Publisher.createIg(Publisher.java:901)<br>\n        at org.hl7.fhir.igtools.publisher.Publisher.execute(Publisher.java:732)<br>\n        at org.hl7.fhir.igtools.publisher.Publisher.main(Publisher.java:8413)</p>\n<p>I see that others are publishing IGs fine, so I'm doing something wrong, but there is no error before the exception is thrown. Anyone else seen this problem?</p>",
        "id": 220112069,
        "sender_full_name": "Daniel Karlsson",
        "timestamp": 1608123676
    },
    {
        "content": "<p>Yes, we have the same problem (local and ci build)</p>",
        "id": 220112314,
        "sender_full_name": "Michaela Ziegler",
        "timestamp": 1608123833
    },
    {
        "content": "<p>Sorry about that (but at least we're not alone).<br>\nSeems to be failing while generating QA HTML (as per the stack trace)...</p>",
        "id": 220112606,
        "sender_full_name": "Daniel Karlsson",
        "timestamp": 1608123999
    },
    {
        "content": "<p>I didn't see this one last night when I built the latest LIVD IG content after updating to the 1.1.53 publisher.</p>",
        "id": 220127435,
        "sender_full_name": "Rob Hausam",
        "timestamp": 1608131546
    },
    {
        "content": "<p><span class=\"user-mention\" data-user-id=\"248736\">@Mark Iantorno</span> Happening for me too.  No changes to the IG and ran with the previous publisher</p>",
        "id": 220146858,
        "sender_full_name": "Lloyd McKenzie",
        "timestamp": 1608140105
    },
    {
        "content": "<p>this means that there's no realm for the IG. I will fix it next release, but there should be a stated realm or region</p>",
        "id": 220149140,
        "sender_full_name": "Grahame Grieve",
        "timestamp": 1608141143
    },
    {
        "content": "<p>Even for non-HL7-published IGs?</p>",
        "id": 220151079,
        "sender_full_name": "Lloyd McKenzie",
        "timestamp": 1608142051
    },
    {
        "content": "<p>My IG <em>does</em> declare a jurisdiction.</p>",
        "id": 220151209,
        "sender_full_name": "Lloyd McKenzie",
        "timestamp": 1608142113
    },
    {
        "content": "<p>they should, yes. Don't have to</p>",
        "id": 220152803,
        "sender_full_name": "Grahame Grieve",
        "timestamp": 1608142852
    },
    {
        "content": "<p>Thanks, is realm = jursidiction</p>",
        "id": 220154230,
        "sender_full_name": "Daniel Karlsson",
        "timestamp": 1608143563
    },
    {
        "content": "<p>yes</p>",
        "id": 220154314,
        "sender_full_name": "Grahame Grieve",
        "timestamp": 1608143595
    },
    {
        "content": "<p>Ok, but my IG has a jurisdiction and is still blowing up...</p>",
        "id": 220155037,
        "sender_full_name": "Lloyd McKenzie",
        "timestamp": 1608143948
    },
    {
        "content": "<p>oh. right. No, it's just a bug, and it'll never work for non-hl7 guides until I release a fix</p>",
        "id": 220155517,
        "sender_full_name": "Grahame Grieve",
        "timestamp": 1608144164
    },
    {
        "content": "<p>I think I am now in this camp too, with an outside IG... still renders okay, just no QA report</p>",
        "id": 220182361,
        "sender_full_name": "John Moehrke",
        "timestamp": 1608157676
    },
    {
        "content": "<p>I tried and it failed to add universal realm to my IG</p>\n<div class=\"codehilite\"><pre><span></span><code>  &lt;jurisdiction&gt;\n    &lt;coding&gt;\n      &lt;!-- This is the code for universal --&gt;\n      &lt;system value=&quot;http://unstats.un.org/unsd/methods/m49/m49.htm&quot;/&gt;\n      &lt;code value=&quot;001&quot;/&gt;\n    &lt;/coding&gt;\n  &lt;/jurisdiction&gt;\n</code></pre></div>",
        "id": 220183403,
        "sender_full_name": "John Moehrke",
        "timestamp": 1608158271
    },
    {
        "content": "<p>what else can I try?</p>",
        "id": 220198428,
        "sender_full_name": "John Moehrke",
        "timestamp": 1608169667
    },
    {
        "content": "<p>nothing</p>",
        "id": 220199371,
        "sender_full_name": "Grahame Grieve",
        "timestamp": 1608170684
    },
    {
        "content": "<p>Version 1.1.54 and 1.1.53  are throwing this null pointer exception. But able to publish using version 1.1.52.</p>",
        "id": 220220810,
        "sender_full_name": "Chetan Jain",
        "timestamp": 1608196759
    },
    {
        "content": "<p>new IG publisher works!</p>",
        "id": 220237463,
        "sender_full_name": "John Moehrke",
        "timestamp": 1608208542
    },
    {
        "content": "<p>Hi all<br>\ni get a NullPointerException building an IG but i can not see where it is failing (v1.1.88):<br>\n....<br>\nGenerating Narratives                                                            (00:09.0568)<br>\nPublishing Content Failed: Cannot invoke \"org.hl7.fhir.r5.renderers.utils.BaseWrappers$PropertyWrapper.hasValues()\" because the return value of \"org.hl7.fhir.r5.renderers.utils.BaseWrappers$BaseWrapper.getChildByName(String)\" is null (00:09.0591)<br>\n                                                                                 (00:09.0592)<br>\nUse -? to get command line help                                                  (00:09.0592)<br>\n                                                                                 (00:09.0592)<br>\nStack Dump (for debugging):                                                      (00:09.0592)<br>\njava.lang.NullPointerException: Cannot invoke \"org.hl7.fhir.r5.renderers.utils.BaseWrappers$PropertyWrapper.hasValues()\" because the return value of \"org.hl7.fhir.r5.renderers.utils.BaseWrappers$BaseWrapper.getChildByName(String)\" is null<br>\n    at org.hl7.fhir.r5.renderers.ResourceRenderer.getPrimitiveValue(ResourceRenderer.java:476)<br>\n    at org.hl7.fhir.r5.renderers.ResourceRenderer.renderResourceHeader(ResourceRenderer.java:402)<br>\n    at org.hl7.fhir.r5.renderers.ProfileDrivenRenderer.render(ProfileDrivenRenderer.java:110)<br>\n    at org.hl7.fhir.r5.renderers.ResourceRenderer.render(ResourceRenderer.java:86)<br>\n    at org.hl7.fhir.igtools.publisher.Publisher.generateNarratives(Publisher.java:1302)<br>\n    at org.hl7.fhir.igtools.publisher.Publisher.loadConformance(Publisher.java:4087)<br>\n    at org.hl7.fhir.igtools.publisher.Publisher.createIg(Publisher.java:941)<br>\n    at org.hl7.fhir.igtools.publisher.Publisher.execute(Publisher.java:793)<br>\n    at org.hl7.fhir.igtools.publisher.Publisher.main(Publisher.java:8927)<br>\n....</p>",
        "id": 262295611,
        "sender_full_name": "Roeland Luykx",
        "timestamp": 1637573192
    },
    {
        "content": "<p>can I reproduce this?</p>",
        "id": 262297443,
        "sender_full_name": "Grahame Grieve",
        "timestamp": 1637574284
    },
    {
        "content": "<p><span class=\"user-mention\" data-user-id=\"191316\">@Grahame Grieve</span> i'am looking for which file it is doing this... but on the error message i can not figure it out. the IG is not public. can i assume that it has something to do with narrative  texxt?</p>",
        "id": 262298632,
        "sender_full_name": "Roeland Luykx",
        "timestamp": 1637575020
    },
    {
        "content": "<p>oh. whoops. An R3 guide?</p>",
        "id": 262299146,
        "sender_full_name": "Grahame Grieve",
        "timestamp": 1637575276
    },
    {
        "content": "<p><span aria-label=\"see no evil\" class=\"emoji emoji-1f648\" role=\"img\" title=\"see no evil\">:see_no_evil:</span> <span aria-label=\"hear no evil\" class=\"emoji emoji-1f649\" role=\"img\" title=\"hear no evil\">:hear_no_evil:</span> yes... its customer decision to have still STU3..</p>",
        "id": 262299273,
        "sender_full_name": "Roeland Luykx",
        "timestamp": 1637575345
    },
    {
        "content": "<p>hmm I don't have a test case set up for that combination. so I'll issue a fix and cross my fingers, though what's wrong in the code is pretty obvious. Should be about 2 hours</p>",
        "id": 262299598,
        "sender_full_name": "Grahame Grieve",
        "timestamp": 1637575550
    },
    {
        "content": "<p><span class=\"user-mention\" data-user-id=\"191316\">@Grahame Grieve</span> good point. with publisher v1.1.84 its working. since 1.1.85 its failing.</p>",
        "id": 262299687,
        "sender_full_name": "Roeland Luykx",
        "timestamp": 1637575603
    }
]